### PR TITLE
feat (risc0) add public input as separate parameter on verification data

### LINKED ISSUE #601: feat (risc0): add public input as separate parameter on verification data
- [x] Aligned cli should take in an public input for risc0
- [ ] Aligned SDK types.rs should serialize public input on commitment
- [ ] Batcher & Operators should verify reported public input matches proof journal public input

### PR DESCRIPTION
**Pull request by taturosati**
- Cli reads pub input for risc0
- Batcher checks pub input matches journal on risc0
- Operator checks pub input matches journal on risc0
- Fibonacci example saves pub input to file

# To test
1. Start everything as normal
2. Run ```make batcher_send_risc0_burst``` and check its verified normal

### CODE REVIEW

File: docs/guides/3_generating_proofs.md [hunk 1]
```diff
@@ -90,4 +90,66 @@ aligned submit \
...
-For more instructions on how to submit proofs, check the [Submitting proofs guide](../guides/0_submitting_proofs.md).
\ No newline at end of file
+For more instructions on how to submit proofs, check the [Submitting proofs guide](../guides/0_submitting_proofs.md).
+
+## Risc0
+
+### Dependencies
+
+This guide assumes that:
+- Risc0 prover installed (instructions [here](https://dev.risczero.com/api/zkvm/quickstart#1-install-the-risc-zero-toolchain))
```
Comment (entropidelic) [inline docs/guides/3_generating_proofs.md]:
```suggestion
- Risc0 toolchain installed (instructions [here](https://dev.risczero.com/api/zkvm/quickstart#1-install-the-risc-zero-toolchain))
```

File: docs/guides/3_generating_proofs.md [hunk 2]
```diff
@@ -90,4 +90,66 @@ aligned submit \
...
+
+### How to generate a proof
+
+First, open the risc0 host file and add the following code to export image id & public input needed by Aligned.
+
+```rust
+fn main() {
+    // your code here
+  
+    std::fs::write("<image_id_file_path>", convert(&<METHOD_ID>))
```
Comment (entropidelic) [inline docs/guides/3_generating_proofs.md]: lets clarify that `image_id_file_path` is the path where you want it to be written. Also, should it be image id or method id? lets unify terminology so its less confusing

File: docs/guides/3_generating_proofs.md [hunk 3]
```diff
@@ -90,4 +90,66 @@ aligned submit \
...
+            .expect("Failed to write fibonacci_id file");
```
Comment (entropidelic) [inline docs/guides/3_generating_proofs.md]: lets not reference a particular programa in the error message

File: batcher/aligned-batcher/src/zk_utils/mod.rs [hunk 4]
```diff
@@ -44,13 +46,17 @@ pub(crate) fn verify(verification_data: &VerificationData) -> bool {
...
             }
-            warn!("Trying to verify Risc0 proof but image ID was not provided. Returning false");
-            false
+
+            let image_id_slice = verification_data.vm_program_code.as_ref().unwrap().as_slice();
+            let public_input = verification_data.pub_input.as_ref().unwrap().as_slice();
+
+            let mut image_id = [0u8; 32];
+            image_id.copy_from_slice(image_id_slice);
+            return verify_risc_zero_proof(verification_data.proof.as_slice(), &image_id, public_input);
```
Comment (entropidelic) [inline batcher/aligned-batcher/src/zk_utils/mod.rs]: small nit but would be ideal if we could catch the vm_program_code and pub_input inside a if let statement and avoid the unwraps

File: docs/guides/3_generating_proofs.md (hunk 3 above)
Comment (taturosati) [reply docs/guides/3_generating_proofs.md]: changed on ef93dfb539db69405643b476a6b8f78e4ccc5dfc

File: docs/guides/3_generating_proofs.md (hunk 2 above)
Comment (taturosati) [reply docs/guides/3_generating_proofs.md]: changed on ef93dfb539db69405643b476a6b8f78e4ccc5dfc

File: batcher/aligned-batcher/src/zk_utils/mod.rs (hunk 4 above)
Comment (srosati) [reply batcher/aligned-batcher/src/zk_utils/mod.rs:59]: I thought checking for none on top and having unwraps below was a bit cleaner than nested ifs, rust won't allow you to have if let _ = .. && let __ = ...

Comment (taturosati) [reply batcher/aligned-batcher/src/zk_utils/mod.rs]: changed on 5deeb9fd7f90f2f0dbe2a16db2551cbd375061ce

**Review by NicolasRampoldi — APPROVED** [review]

**Review by JuArce — CHANGES_REQUESTED** [review]
There is missing update the risc0 guide in `0_submitting_proofs`

File: batcher/aligned-batcher/src/zk_utils/mod.rs [hunk 5]
```diff
@@ -44,12 +46,26 @@ pub(crate) fn verify(verification_data: &VerificationData) -> bool {
...
-                return verify_risc_zero_proof(verification_data.proof.as_slice(), &image_id);
+                return verify_risc_zero_proof(
+                    verification_data.proof.as_slice(),
+                    &image_id,
+                    pub_input,
+                );
             }
-            warn!("Trying to verify Risc0 proof but image ID was not provided. Returning false");
+
+            warn!("Trying to verify Risc0 proof but image id or public input was not provided. Returning false");
```
Comment (JuArce) [inline batcher/aligned-batcher/src/zk_utils/mod.rs:62]: Why is this warn duplicated?

**Review by entropidelic — APPROVED** [review]

**Comment by entropidelic** [comment]
let's handle the changes that @JuArce requested and then it can be merged

**Review by JuArce — APPROVED** [review]

**Pull request merged**